extensions/vscode/src/code-lens/on-request-code-lens.ts

```diff
@@ -0,0 +1,119 @@
+import {
+  CodeLens,
+  CodeLensProvider,
+  Event,
+  EventEmitter,
+  Position,
+  ProviderResult,
+  TextDocument,
+  workspace,
+} from "vscode";
+import { nearestDartFrogProject } from "../utils";
+import path = require("path");
+
+abstract class ConfigurableCodeLensProvider implements CodeLensProvider {
+  protected codeLenses: CodeLens[] = [];
+
+  protected _onDidChangeCodeLenses: EventEmitter<void> =
+    new EventEmitter<void>();
+  public readonly onDidChangeCodeLenses: Event<void> =
+    this._onDidChangeCodeLenses.event;
+
+  constructor() {
+    workspace.onDidChangeConfiguration(() => {
+      this._onDidChangeCodeLenses.fire();
+    });
+  }
+
+  // eslint-disable-next-line no-unused-vars
+  public provideCodeLenses(document: TextDocument): ProviderResult<CodeLens[]> {
+    if (!this.hasEnabledCodeLenses()) {
+      return undefined;
+    }
+    return this.codeLenses;
+  }
+
+  public resolveCodeLens?(
+    codeLens: CodeLens
+    // eslint-disable-next-line no-unused-vars
```

Comment:
Is this comment in the right line? the next line is the method return type, and from what I can tell, there are no unused variables